dask_saturn/core.py

```diff
@@ -100,66 +86,106 @@ def scheduler_address(self):
     def dashboard_link(self):
         return self._dashboard_link
 
+    def __await__(self):
+        async def _():
+            if self.status == "created":
+                await self._start()
+            return self
+
+        return _().__await__()
+
     @property
     def scheduler_info(self):
+        if self.cluster_url is None:
+            return {"workers": {}}
         url = urljoin(self.cluster_url, "scheduler_info")
         response = requests.get(url, headers=HEADERS)
         if not response.ok:
-            if self._get_pod_status() in ["error", "closed", "stopped"]:
+            self._refresh_status()
+            if self.status in ["error", "closed", "stopped"]:
                 for pc in self.periodic_callbacks.values():
                     pc.stop()
+            if self.status == "error":
```

Comment:
since these statuses are being repeating in a few places, I think they should be concentrated in constants at the top of the module